include/envoy/server/instance.h

```diff
@@ -188,6 +188,11 @@ class Instance {
    * @return information about the local environment the server is running in.
    */
   virtual const LocalInfo::LocalInfo& localInfo() PURE;
+
+  /**
+   * @return the server-wide cluster manager factory.
+   */
+  virtual Upstream::ClusterManagerFactory& clusterManagerFactory() PURE;
```

Comment:
I think I see why this was done. Can we avoid adding this to the interface (and all the mocks, etc.) by just passing the factory to the ConfigurationImpl constructor?